sink/sink

```diff
@@ -89,9 +97,31 @@ def expand(val, env):
     else:
         return val
 
+# Compare two JSON style values to see if one is contained in the other
+# For lists and dict make sure all items in expect are also present in result
+# For any other values a strict comparison is used
+def contains(expect, result):
+    if isinstance(expect, list) and isinstance(result, list):
+        for i in range(0, len(expect)):
+            res = result[i] if i < len(result) else None
+            if not contains(expect[i], res):
+                return False
+        return True
```

Comment:
This will say False if the result list does not start with the expect list, but the documentation suggests that expect should merely be *anywhere* in result. I. e. `contains([2], [1, 2, 3])` will be `False` with this. So either the description or the algorithm needs to be changed.

Is there really a use case for comparing GitHub states where the "expected list is a prefix of result" behaviour makes sense? It seems to me that comparing for equality or real "contained in" makes more sense and is more robust against future changes in GitHub's API?